bin_heap.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+
+
+class BinaryHeap(object):
+    def __init__(self):
+        self.heaplist = [0]
+        self.currentsize = 0
+
+    def push(self, val):
+        self.heaplist.append(val)
+        self.currentsize += 1
+        self.bubble_up(self.currentsize)
+
+    def pop(self):
+        return_val = self.heaplist[1]
```

Comment:
When the heap is "empty", this raises an index error, as expected, but it looks odd because the index requested is `1`.  This means that it is clear to the user that although the heap is reporting itself as empty it really has one thing in it.  That's odd.